cura/LayerDataBuilder.py

```diff
@@ -48,28 +48,63 @@ def setLayerThickness(self, layer, thickness):
 
         self._layers[layer].setThickness(thickness)
 
-    def build(self):
+    #   material color map: [r, g, b, a] for each extruder row.
+    #   line_type_brightness: compatibility layer view uses line type brightness of 0.5
+    def build(self, material_color_map, line_type_brightness = 1.0):
         vertex_count = 0
         index_count = 0
         for layer, data in self._layers.items():
             vertex_count += data.lineMeshVertexCount()
             index_count += data.lineMeshElementCount()
 
         vertices = numpy.empty((vertex_count, 3), numpy.float32)
+        line_dimensions = numpy.empty((vertex_count, 2), numpy.float32)
         colors = numpy.empty((vertex_count, 4), numpy.float32)
         indices = numpy.empty((index_count, 2), numpy.int32)
+        extruders = numpy.empty((vertex_count), numpy.int32)
+        line_types = numpy.empty((vertex_count), numpy.int32)
 
         vertex_offset = 0
         index_offset = 0
         for layer, data in self._layers.items():
-            ( vertex_offset, index_offset ) = data.build( vertex_offset, index_offset, vertices, colors, indices)
+            ( vertex_offset, index_offset ) = data.build( vertex_offset, index_offset, vertices, colors, line_dimensions, extruders, line_types, indices)
             self._element_counts[layer] = data.elementCount
 
         self.addVertices(vertices)
+        colors[:, 0:3] *= line_type_brightness
         self.addColors(colors)
         self.addIndices(indices.flatten())
 
+        material_colors = numpy.zeros((line_dimensions.shape[0], 4), dtype=numpy.float32)
+        for extruder_nr in range(material_color_map.shape[0]):
+            material_colors[extruders == extruder_nr] = material_color_map[extruder_nr]
+        material_colors[line_types == LayerPolygon.MoveCombingType] = colors[line_types == LayerPolygon.MoveCombingType]
```

Comment:
As just discussed; Numpy arrays work differently. Please add a comment here to indicate that.